ccl/changefeedccl: TestChangefeedDistributionStrategy failed

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/474277c7-a2a4-482f-93e1-f5730939539c?testReportRun=23&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [097880977a1f5bfb8377afd2a407cdf7ca4fe377](https://github.com/cockroachdb/cockroach/commits/097880977a1f5bfb8377afd2a407cdf7ca4fe377):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy2647128566
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy2647128566
--- FAIL: TestChangefeedDistributionStrategy (389.90s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    changefeed_dist_test.go:443: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:443
        	Error:      	Not equal: 
        	            	expected: 65
        	            	actual  : 64
        	Test:       	TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (119.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34809

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/9ffd9f45-b078-46cf-9a2b-c51d378af62d?testReportRun=12&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [ff291e3ee70eedac50b985860502d2f0b0867003](https://github.com/cockroachdb/cockroach/commits/ff291e3ee70eedac50b985860502d2f0b0867003):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy3048139665
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy3048139665
--- FAIL: TestChangefeedDistributionStrategy (492.55s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    changefeed_dist_test.go:443: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:443
        	Error:      	Not equal: 
        	            	expected: 65
        	            	actual  : 64
        	Test:       	TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (112.12s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/9951f134-a4c7-4428-91a7-32a3cf81e8cd?testReportRun=1&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [d2d974202781321ddb4fb1115c68fe7566ab9d06](https://github.com/cockroachdb/cockroach/commits/d2d974202781321ddb4fb1115c68fe7566ab9d06):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy1600664232
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy1600664232
--- FAIL: TestChangefeedDistributionStrategy (4104.26s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    changefeed_dist_test.go:287: error executing 'CREATE TABLE x (id INT PRIMARY KEY)': sql: database is closed
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (662.48s)
=== RUN   TestChangefeedDistributionStrategy/locality=none,distribution=none
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    changefeed_dist_test.go:368: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:368
        	Error:      	Should be true
        	Test:       	TestChangefeedDistributionStrategy/locality=none,distribution=none
        	Messages:   	unexpected counts [64 0 0 0 0 0 0 0], partitions: [{1 /Tenant/10/Table/104/{1-2}}]
    --- FAIL: TestChangefeedDistributionStrategy/locality=none,distribution=none (1616.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/bde6ea97-5668-445e-be2c-c0568d03b4c1?testReportRun=27&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy3325216572
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy3325216572
--- FAIL: TestChangefeedDistributionStrategy (396.16s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    changefeed_dist_test.go:443: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:443
        	Error:      	Not equal: 
        	            	expected: 65
        	            	actual  : 64
        	Test:       	TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (84.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=27</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/f4b3f5ba-52a2-4e6f-bc32-2aa9e77eb2b9?testReportRun=3&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):

Fatal error:

```
panic: test timed out after 1h29m52s
running tests:
	TestChangefeedDistributionStrategy (1h29m43s)
	TestChangefeedDistributionStrategy/locality=constrained,distribution=none (24m37s)

```
Stack: 

```
goroutine 870889930 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy2701874097
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestChangefeedDistributionStrategy/locality=none,distribution=none
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster.go:488: migration-job-find-already-completed: failed to connect to n2 at 127.0.0.1:45063: grpc: refusing to dial; node is quiescing [code 7/PermissionDenied]
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=none
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/10359a57-17c3-4f96-95b8-f8685fb95d35?testReportRun=22&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy2940480936
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy2940480936
--- FAIL: TestChangefeedDistributionStrategy (489.33s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    changefeed_dist_test.go:443: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:443
        	Error:      	Not equal: 
        	            	expected: 65
        	            	actual  : 64
        	Test:       	TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (97.32s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/6b1dfd63-2f2d-4df6-818e-e83da6708e4c?testReportRun=3&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [53e4efd389167110d687e5dea8bf915fe2582a25](https://github.com/cockroachdb/cockroach/commits/53e4efd389167110d687e5dea8bf915fe2582a25):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy214467434
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestChangefeedDistributionStrategy/locality=none,distribution=none
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster.go:488: polling for queued jobs to complete: poll-show-jobs: node unavailable; try another peer
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=none
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestChangefeedDistributionStrategy/locality=none,distribution=balanced_simple
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    changefeed_dist_test.go:287: error executing 'ALTER TABLE x SPLIT AT SELECT id FROM x WHERE id > 0': dial tcp 127.0.0.1:40075: connect: connection refused
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/9da80718-4636-4bcb-bc9a-b35e2374de6f?testReportRun=3&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [4f31fbae529e459c00ef2490e193c0c246e8d0f9](https://github.com/cockroachdb/cockroach/commits/4f31fbae529e459c00ef2490e193c0c246e8d0f9):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy3347662761
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy3347662761
--- FAIL: TestChangefeedDistributionStrategy (480.51s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    changefeed_dist_test.go:443: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:443
        	Error:      	Not equal: 
        	            	expected: 65
        	            	actual  : 64
        	Test:       	TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (113.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=3</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/b5c856ee-5ca5-4bba-94c6-bfdd5a69aa78?testReportRun=8&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy3823007362
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy3823007362
--- FAIL: TestChangefeedDistributionStrategy (396.80s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    changefeed_dist_test.go:443: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:443
        	Error:      	Not equal: 
        	            	expected: 65
        	            	actual  : 64
        	Test:       	TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (79.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/3ad3fbcb-6df0-4bd9-b83c-73c73250de72?testReportRun=1&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy3136151353
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy3136151353
--- FAIL: TestChangefeedDistributionStrategy (3113.68s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=none
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    sql_runner.go:87: error executing query="CREATE CHANGEFEED FOR x INTO 'null://' WITH initial_scan='no', execution_locality='y=1'" args=[]: dial tcp 127.0.0.1:33071: connect: connection refused
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=none (1946.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/e3897ad1-3bd1-4605-888d-d6535e862765?testReportRun=3&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy2364044652
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy2364044652
--- FAIL: TestChangefeedDistributionStrategy (1458.50s)
=== RUN   TestChangefeedDistributionStrategy/locality=none,distribution=balanced_simple
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster.go:488: polling for queued jobs to complete: poll-show-jobs: node unavailable; try another peer
    --- FAIL: TestChangefeedDistributionStrategy/locality=none,distribution=balanced_simple (339.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/8b9060b1-d171-4114-873f-30388b68e556?testReportRun=3&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):

Fatal error:

```
panic: test timed out after 1h29m52s
running tests:
	TestChangefeedDistributionStrategy (1h29m44s)
	TestChangefeedDistributionStrategy/locality=constrained,distribution=none (59m5s)

```
Stack: 

```
goroutine 822447945 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy3719872864
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=none
    test_server_shim.go:168: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/eba359e4-5132-4d71-874d-9ee5c794259d?testReportRun=6&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy2520313089
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy2520313089
--- FAIL: TestChangefeedDistributionStrategy (432.45s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    changefeed_dist_test.go:443: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:443
        	Error:      	Not equal: 
        	            	expected: 65
        	            	actual  : 64
        	Test:       	TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (78.93s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/211f9e01-1463-4c98-87a1-aeac9353d344?testReportRun=3&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c316d6a615fa02a05357a20bf03b8a197ab27810](https://github.com/cockroachdb/cockroach/commits/c316d6a615fa02a05357a20bf03b8a197ab27810):


```
=== RUN   TestChangefeedDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedDistributionStrategy1734773040
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:445: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedDistributionStrategy1734773040
--- FAIL: TestChangefeedDistributionStrategy (3934.97s)
=== RUN   TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    testcluster.go:488: polling for queued jobs to complete: poll-show-jobs: node unavailable; try another peer
    --- FAIL: TestChangefeedDistributionStrategy/locality=constrained,distribution=balanced_simple (675.60s)
=== RUN   TestChangefeedDistributionStrategy/locality=none,distribution=none
    test_server_shim.go:168: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    changefeed_dist_test.go:368: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:368
        	Error:      	Should be true
        	Test:       	TestChangefeedDistributionStrategy/locality=none,distribution=none
        	Messages:   	unexpected counts [0 36 0 0 0 0 0 28], partitions: [{2 /Tenant/10/Table/104/1{-/6}, /Tenant/10/Table/104/1/1{2-3}, /Tenant/10/Table/104/1/1{4-7}, /Tenant/10/Table/104/1/1{8-9}, /Tenant/10/Table/104/1/2{0-1}, /Tenant/10/Table/104/1/2{3-4}, /Tenant/10/Table/104/1/2{6-8}, /Tenant/10/Table/104/1/{29-30}, /Tenant/10/Table/104/1/3{1-3}, /Tenant/10/Table/104/1/3{5-6}, /Tenant/10/Table/104/1/3{7-8}, /Tenant/10/Table/104/1/{39-40}, /Tenant/10/Table/104/1/4{1-3}, /Tenant/10/Table/104/1/4{4-5}, /Tenant/10/Table/104/1/4{7-8}, /Tenant/10/Table/104/1/{49-51}, /Tenant/10/Table/104/1/5{2-3}, /Tenant/10/Table/104/1/5{5-6}, /Tenant/10/Table/104/{1/57-2}} {8 /Tenant/10/Table/104/1/{6-12}, /Tenant/10/Table/104/1/1{3-4}, /Tenant/10/Table/104/1/1{7-8}, /Tenant/10/Table/104/1/{19-20}, /Tenant/10/Table/104/1/2{1-3}, /Tenant/10/Table/104/1/2{4-6}, /Tenant/10/Table/104/1/2{8-9}, /Tenant/10/Table/104/1/3{0-1}, /Tenant/10/Table/104/1/3{3-5}, /Tenant/10/Table/104/1/3{6-7}, /Tenant/10/Table/104/1/3{8-9}, /Tenant/10/Table/104/1/4{0-1}, /Tenant/10/Table/104/1/4{3-4}, /Tenant/10/Table/104/1/4{5-7}, /Tenant/10/Table/104/1/4{8-9}, /Tenant/10/Table/104/1/5{1-2}, /Tenant/10/Table/104/1/5{3-5}, /Tenant/10/Table/104/1/5{6-7}}]
    --- FAIL: TestChangefeedDistributionStrategy/locality=none,distribution=none (994.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/changefeedccl/changefeed_dist_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_dist_test.go
@@ -223,6 +223,7 @@ func (rdt *rangeDistributionTester) getPartitions() (partitions []sql.SpanPartit
 			return errors.New("no partitions found")
 		}
 	})
+	rdt.t.Logf("found partitions: %v", partitions)
 	return
 }
 
@@ -330,8 +331,8 @@ func (rdt *rangeDistributionTester) countRangesPerNode(partitions []sql.SpanPart
 				}
 			}
 		}
-
 	}
+	rdt.t.Logf("range counts: %v", counts)
 	return counts
 }
 
@@ -345,10 +346,9 @@ func TestChangefeedDistributionStrategy(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	// The test is slow.
+	// The test is slow and will time out under deadlock/race/stress.
 	skip.UnderShort(t)
-	skip.UnderStress(t)
-	skip.UnderRace(t)
+	skip.UnderDuress(t)
 
 	noLocality := func(i int) []roachpb.Tier {
 		return make([]roachpb.Tier, 0)

